include/chopper/data_store.hpp

```diff
@@ -8,14 +8,38 @@
 #include <chopper/helper.hpp>
 #include <chopper/layout/hibf_statistics.hpp>
 #include <chopper/layout/layout.hpp>
-#include <chopper/layout/previous_level.hpp>
 #include <chopper/sketch/toolbox.hpp>
 
 namespace chopper
 {
 
 struct data_store
 {
+    /*!\brief Stores information of the previous level of a given IBF.
+     *
+     * When computing a hierarchical layout, the `data_store` data structure is used with local copies for each IBF
+     * within the hierarchical structure of the HIBF. To keep track of the hierarchy, the `previous_level` stores
+     * information about the previous level (where the corresponding merged bin is located).
+     */
+    struct previous_level
+    {
+        std::vector<size_t> bin_indices{};
+        std::string bin_indices_str;
+        std::string num_of_bins;
+        std::string estimated_sizes;
+        std::string optimal_score;
+        std::string correction;
+        std::string tmax;
+
+        bool empty() const
+        {
+            assert(((((bin_indices.empty() == num_of_bins.empty()) == estimated_sizes.empty()) == optimal_score.empty())
+                    == correction.empty())
+                   == tmax.empty());
```

Comment:
```suggestion
            assert(bin_indices.empty() == num_of_bins.empty());
            assert(num_of_bins.empty() == estimated_sizes.empty());
            assert(estimated_sizes.empty() == optimal_score.empty());
            assert(optimal_score.empty() == correction.empty());
            assert(correction.empty() == tmax.empty());
```
Maybe that's easier?
The old assert kinda got out of hand...

Reply:
you are right! :laughing: